### ISSUE TITLE
v5.2.4 TiDB OOM during tpcc workload

### ISSUE THREAD

**Issue by dbsid** [pingcap/tidb#34354]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

restore 1k warehouse tpcc data
run benchmarksql client with 600 threads for 3 days

### 2. What did you expect to see? (Required)
no TiDB OOM

### 3. What did you see instead (Required)
Application failure by TiDB OOM, seems the oom is causing by auto analyze failure

BenchmarkSQL Client Error
```
Term-00, Running Average tpmTOTAL: 190512.40000000002    Current tpmTOTAL: 4212837120    Memory Usage: 66MB / 101MB
20:25:03,796 [Thread-34] ERROR  jTPCCTData : Unexpected SQLException in PAYMENT
20:25:03,798 [Thread-16] ERROR  jTPCCTData : Unexpected SQLException in NEW_ORDER
20:25:03,798 [Thread-1] ERROR  jTPCCTData : Unexpected SQLException in DELIVERY_BG
20:25:03,798 [Thread-43] ERROR  jTPCCTData : Unexpected SQLException in NEW_ORDER
20:25:03,804 [Thread-43] ERROR  jTPCCTData : Communications link failure
The last packet successfully received from the server was 1,752 milliseconds ago.  The last packet sent successfully to the server was 1,752 milliseconds ago.
20:25:03,797 [Thread-37] ERROR  jTPCCTData : Unexpected SQLException in PAYMENT
20:25:03,805 [Thread-37] ERROR  jTPCCTData : Communications link failure
The last packet successfully received from the server was 1,743 milliseconds ago.  The last packet sent successfully to the server was 1,743 milliseconds ago.
20:25:03,796 [Thread-49] ERROR  jTPCCTData : Unexpected SQLException in PAYMENT
20:25:03,806 [Thread-49] ERROR  jTPCCTData : Communications link failure
The last packet successfully received from the server was 1,751 milliseconds ago.  The last packet sent successfully to the server was 1,751 milliseconds ago.
20:25:03,806 [Thread-22] FATAL  jTPCCTerminal : Unexpected SQLException on rollback: No operations allowed after connection closed.
20:25:03,798 [Thread-10] ERROR  jTPCCTData : Unexpected SQLException in NEW_ORDER
20:25:03,808 [Thread-10] ERROR  jTPCCTData : Communications link failure
The last packet successfully received from the server was 1,753 milliseconds ago.  The last packet sent successfully to the server was 1,753 milliseconds ago.
```

dmesg -T
```
[root@localhost log]# dmesg -T | grep -i oom
[一 5月  2 20:24:57 2022] tidb-server invoked oom-killer: gfp_mask=0x14200ca(GFP_HIGHUSER_MOVABLE), nodemask=0,  order=0, oom_score_adj=0
[一 5月  2 20:24:57 2022] [<ffff000008210ca8>] oom_kill_process+0x2b8/0x524
[一 5月  2 20:24:57 2022] [ pid ]   uid  tgid total_vm      rss nr_ptes nr_pmds swapents oom_score_adj name
[一 5月  2 20:24:58 2022] oom_reaper: reaped process 34462 (tidb-server), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

tidb.log
```
[2022/05/02 17:14:14.468 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 16:54:23.177 +0800 CST, GC safe point is 2022-05-02 17:04:04.477 +0800 CST"]
[2022/05/02 17:14:14.468 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=19m51.272808855s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 16:54:23.177 +0800 CST, GC safe point is 2022-05-02 17:04:04.477 +0800 CST"]
[2022/05/02 17:34:14.676 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:14:14.48 +0800 CST, GC safe point is 2022-05-02 17:24:04.528 +0800 CST"]
[2022/05/02 17:34:14.676 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=20m0.196398495s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:14:14.48 +0800 CST, GC safe point is 2022-05-02 17:24:04.528 +0800 CST"]
[2022/05/02 17:54:25.438 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:34:14.677 +0800 CST, GC safe point is 2022-05-02 17:44:04.527 +0800 CST"]
[2022/05/02 17:54:25.438 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=20m10.757251117s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:34:14.677 +0800 CST, GC safe point is 2022-05-02 17:44:04.527 +0800 CST"]
[2022/05/02 18:14:25.346 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:54:25.427 +0800 CST, GC safe point is 2022-05-02 18:04:04.527 +0800 CST"]
[2022/05/02 18:14:25.346 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=19m59.906543184s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 17:54:25.427 +0800 CST, GC safe point is 2022-05-02 18:04:04.527 +0800 CST"]
[2022/05/02 18:34:23.644 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:16:18.527 +0800 CST, GC safe point is 2022-05-02 18:24:04.527 +0800 CST"]
[2022/05/02 18:34:23.645 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=18m5.107283516s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:16:18.527 +0800 CST, GC safe point is 2022-05-02 18:24:04.527 +0800 CST"]
[2022/05/02 18:54:25.254 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:34:23.627 +0800 CST, GC safe point is 2022-05-02 18:44:04.577 +0800 CST"]
[2022/05/02 18:54:25.254 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=20m1.607394701s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:34:23.627 +0800 CST, GC safe point is 2022-05-02 18:44:04.577 +0800 CST"]
[2022/05/02 19:14:26.778 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:54:25.228 +0800 CST, GC safe point is 2022-05-02 19:04:04.478 +0800 CST"]
[2022/05/02 19:14:26.778 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=20m1.518806228s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 18:54:25.228 +0800 CST, GC safe point is 2022-05-02 19:04:04.478 +0800 CST"]
[2022/05/02 19:34:26.393 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:14:26.778 +0800 CST, GC safe point is 2022-05-02 19:24:04.577 +0800 CST"]
[2022/05/02 19:34:26.394 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=19m59.604340834s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:14:26.778 +0800 CST, GC safe point is 2022-05-02 19:24:04.577 +0800 CST"]
[2022/05/02 19:54:18.041 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:34:48.277 +0800 CST, GC safe point is 2022-05-02 19:44:04.477 +0800 CST"]
[2022/05/02 19:54:18.041 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=19m29.733437993s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:34:48.277 +0800 CST, GC safe point is 2022-05-02 19:44:04.477 +0800 CST"]
[2022/05/02 20:14:18.885 +08:00] [ERROR] [analyze.go:146] ["analyze failed"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:54:18.027 +0800 CST, GC safe point is 2022-05-02 20:04:04.527 +0800 CST"]
[2022/05/02 20:14:18.885 +08:00] [ERROR] [update.go:1083] ["[stats] auto analyze failed"] [sql="analyze table %n.%n"] [cost_time=20m0.83448876s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-05-02 19:54:18.027 +0800 CST, GC safe point is 2022-05-02 20:04:04.527 +0800 CST"]
[2022/05/02 20:25:18.970 +08:00] [INFO] [printer.go:33] ["Welcome to TiDB."] ["Release Version"=v5.2.4] [Edition=Community] ["Git Commit Hash"=fe5aa9760b587799361c954cffab4d786dbdbf7e] ["Git Branch"=heads/refs/tags/v5.2.4] ["UTC Build Time"="2022-04-18 06:20:02"] [GoVersion=go1.16.4] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=v3.0.0-60965b006877ca7234adaced7890d7b029ed1306]
[2022/05/02 20:25:18.970 +08:00] [INFO] [trackerRecorder.go:28] ["Mem Profile Tracker started"]
[2022/05/02 20:25:18.971 +08:00] [INFO] [printer.go:47] ["loaded config"] [config="{\"host\":\"0.0.0.0\",\"advertise-address\":\"172.16.4.106\",\"port\":4000,\"cors\":\"\",\"store\":\"tikv\",\"path\":\"172.16.4.104:2379\",\"socket\":\"\",\"lease\":\"45s\",\"run-ddl\":true,\"split-table\":true,\"token-limit\":1000,\"oom-use-tmp-storage\":true,\"tmp-storage-path\":\"/tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage\",\"oom-action\":\"cancel\",\"mem-quota-query\":1073741824,\"tmp-storage-quota\":-1,\"enable-batch-dml\":false,\"lower-case-table-names\":2,\"server-version\":\"\",\"log\":{\"level\":\"error\",\"format\":\"text\",\"disable-timestamp\":null,\"enable-timestamp\":null,\"disable-error-stack\":null,\"enable-error-stack\":null,\"file\":{\"filename\":\"/data1/v52/tidb-4000/log/tidb.log\",\"max-size\":300,\"max-days\":0,\"max-backups\":0},\"enable-slow-log\":true,\"slow-query-file\":\"/data1/v52/tidb-4000/log/tidb_slow_query.log\",\"slow-threshold\":300,\"expensive-threshold\":10000,\"query-log-max-len\":4096,\"record-plan-in-slow-log\":1},\"security\":{\"skip-grant-table\":false,\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"require-secure-transport\":false,\"cluster-ssl-ca\":\"\",\"cluster-ssl-cert\":\"\",\"cluster-ssl-key\":\"\",\"cluster-verify-cn\":null,\"spilled-file-encryption-method\":\"plaintext\",\"enable-sem\":false,\"auto-tls\":false},\"status\":{\"status-host\":\"0.0.0.0\",\"metrics-addr\":\"\",\"status-port\":10080,\"metrics-interval\":15,\"report-status\":true,\"record-db-qps\":false},\"performance\":{\"max-procs\":0,\"max-memory\":0,\"server-memory-quota\":0,\"memory-usage-alarm-ratio\":0.8,\"stats-lease\":\"3s\",\"stmt-count-limit\":5000,\"feedback-probability\":0,\"query-feedback-limit\":512,\"pseudo-estimate-ratio\":0.8,\"force-priority\":\"NO_PRIORITY\",\"bind-info-lease\":\"3s\",\"txn-entry-size-limit\":6291456,\"txn-total-size-limit\":104857600,\"tcp-keep-alive\":true,\"tcp-no-delay\":true,\"cross-join\":true,\"run-auto-analyze\":true,\"distinct-agg-push-down\":false,\"committer-concurrency\":128,\"max-txn-ttl\":3600000,\"mem-profile-interval\":\"1m\",\"index-usage-sync-lease\":\"0s\",\"gogc\":100,\"enforce-mpp\":false},\"prepared-plan-cache\":{\"enabled\":true,\"capacity\":100,\"memory-guard-ratio\":0.1},\"opentracing\":{\"enable\":false,\"rpc-metrics\":false,\"sampler\":{\"type\":\"const\",\"param\":1,\"sampling-server-url\":\"\",\"max-operations\":0,\"sampling-refresh-interval\":0},\"reporter\":{\"queue-size\":0,\"buffer-flush-interval\":0,\"log-spans\":false,\"local-agent-host-port\":\"\"}},\"proxy-protocol\":{\"networks\":\"\",\"header-timeout\":5},\"pd-client\":{\"pd-server-timeout\":3},\"tikv-client\":{\"grpc-connection-count\":4,\"grpc-keepalive-time\":10,\"grpc-keepalive-timeout\":3,\"grpc-compression-type\":\"none\",\"commit-timeout\":\"41s\",\"async-commit\":{\"keys-limit\":256,\"total-key-size-limit\":4096,\"safe-window\":2000000000,\"allowed-clock-drift\":500000000},\"max-batch-size\":128,\"overload-threshold\":200,\"max-batch-wait-time\":0,\"batch-wait-size\":8,\"enable-chunk-rpc\":true,\"region-cache-ttl\":600,\"store-limit\":0,\"store-liveness-timeout\":\"1s\",\"copr-cache\":{\"capacity-mb\":1000},\"ttl-refreshed-txn-size\":33554432},\"binlog\":{\"enable\":false,\"ignore-error\":false,\"write-timeout\":\"15s\",\"binlog-socket\":\"\",\"strategy\":\"range\"},\"compatible-kill-query\":false,\"plugin\":{\"dir\":\"\",\"load\":\"\"},\"pessimistic-txn\":{\"max-retry-count\":256,\"deadlock-history-capacity\":10,\"deadlock-history-collect-retryable\":false},\"check-mb4-value-in-utf8\":true,\"max-index-length\":3072,\"index-limit\":64,\"table-column-count-limit\":1017,\"graceful-wait-before-shutdown\":0,\"alter-primary-key\":false,\"treat-old-version-utf8-as-utf8mb4\":true,\"enable-table-lock\":false,\"delay-clean-table-lock\":0,\"split-region-max-num\":1000,\"stmt-summary\":{\"enable\":true,\"enable-internal-query\":false,\"max-stmt-count\":3000,\"max-sql-length\":4096,\"refresh-interval\":1800,\"history-size\":24},\"top-sql\":{\"receiver-address\":\"\"},\"repair-mode\":false,\"repair-table-list\":[],\"isolation-read\":{\"engines\":[\"tikv\",\"tiflash\",\"tidb\"]},\"max-server-connections\":0,\"new_collations_enabled_on_first_bootstrap\":false,\"experimental\":{\"allow-expression-index\":false},\"enable-collect-execution-info\":true,\"skip-register-to-dashboard\":false,\"enable-telemetry\":true,\"labels\":{},\"enable-global-index\":false,\"deprecate-integer-display-length\":false,\"enable-enum-length-limit\":true,\"stores-refresh-interval\":60,\"enable-tcp4-only\":false,\"enable-forwarding\":false}"]
```

![7kz6fNNhlH](https://user-images.githubusercontent.com/2316425/166236668-9942d3ef-7c30-4a1a-8442-e0d1408443f3.png)
![origin_img_v2_8d252e62-553c-40b8-8c45-eabf48e5e88g](https://user-images.githubusercontent.com/2316425/166236693-cc606f13-60de-445b-a28f-bb7f1244ad6d.png)
![0DvImx8Mtt](https://user-images.githubusercontent.com/2316425/166236710-85150c6a-9582-4e0d-bda3-91621ec54aed.png)

### 4. What is your TiDB version? (Required)
v5.2.4

**Comment by chrysan** [comment]
It's a known issue which is not merged into v5.2+ branch, https://github.com/pingcap/tidb/pull/29306 will triage it and make it available in v5.2.5.

**Comment by chrysan** [comment]
https://github.com/pingcap/tidb/pull/29306 is triaged and merged. Fix will be available in v5.2.5, so close this issue.
